src/xdsl/dialects/llvm.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+from dataclasses import dataclass
+
+from xdsl.irdl import *
+from xdsl.ir import *
+from typing import overload
+from xdsl.dialects.builtin import StringAttr, ArrayOfConstraint
+
+if TYPE_CHECKING:
+    from xdsl.parser import Parser
+    from xdsl.printer import Printer
+
+
+@dataclass
+class LLVM:
+    ctx: MLContext
+
+    def __post_init__(self):
+        self.ctx.register_attr(LLVMStructType)
+
+        self.ctx.register_op(LLVMExtractValue)
+        self.ctx.register_op(LLVMInsertValue)
+
+
+@irdl_attr_definition
+class LLVMStructType(ParametrizedAttribute):
+    name = "llvm.struct"
+
+    struct_name = ParameterDef(StringAttr)
+    types = ParameterDef(ArrayOfConstraint(AnyAttr()))
```

Comment:
I think you are on an older xDSL version.
The new way of defining attributes is
```
struct_name: StringAttr
types: ArrayAttr[Attribute]
```
if I'm correct